app/src/org/commcare/dalvik/activities/EntityDetailActivity.java

```diff
@@ -81,6 +86,14 @@ public void onCreate(Bundle savedInstanceState) {
         
         Vector<Entry> entries = session.getEntriesForCommand(passedCommand == null ? session.getCommand() : passedCommand);
         prototype = entries.elementAt(0);
+
+        // (We shouldn't need the "" here, but we're avoiding making changes to
+        // commcare core for release issues)
+        if (entries.size() == 1 && (prototype.getXFormNamespace() == null || prototype.getXFormNamespace().equals(""))) {

```

Comment:
We do this calculation in two places, right? Seems like it's complex enough that 's dangerous to not centralize that it possible? I know that applies to a couple parts of the Awesome/normal display, but this in particular feels very fragile to recompute in two places.